src/theory/arith/nl/iand_solver.cpp

```diff
@@ -215,17 +200,7 @@ Node IAndSolver::mkIOr(unsigned k, Node x, Node y) const
 Node IAndSolver::mkINot(unsigned k, Node x) const
 {
   NodeManager* nm = NodeManager::currentNM();
-  Node ret = nm->mkNode(MINUS, twoToKMinusOne(k), x);
-  ret = Rewriter::rewrite(ret);
-  return ret;
-}
-
-Node IAndSolver::iextract(unsigned i, unsigned j, Node n) const
-{
-  NodeManager* nm = NodeManager::currentNM();
-  //  ((_ extract i j) n) is n / 2^j mod 2^{i-j+1}
-  Node n2j = nm->mkNode(INTS_DIVISION_TOTAL, n, twoToK(j));
-  Node ret = nm->mkNode(INTS_MODULUS_TOTAL, n2j, twoToK(i - j + 1));
+  Node ret = nm->mkNode(MINUS, d_iandTable.twoToKMinusOne(k), x);
```

Comment:
Maybe this function can also go to the helper file `iand_table`.

Reply:
Just to clarify, are you thinking that all the `mk*` methods that are currently in `iand_solver` should go into the common file?